src/dscanner/analysis/properly_documented_public_functions.d

```diff
@@ -187,36 +187,81 @@ private:
 		return comment;
 	}
 
-	void checkDdocParams(size_t line, size_t column, const Parameters params)
+	void checkDdocParams(size_t line, size_t column, const Parameters params,
+					     const TemplateParameters templateParameters = null)
 	{
-		import std.algorithm.searching : canFind;
+		import std.array : array;
+		import std.algorithm.searching : canFind, countUntil;
+		import std.algorithm.iteration : map;
+		import std.algorithm.mutation : remove;
+		import std.range : indexed, iota;
+
+		// convert templateParameters into a string[] for faster access
+		const(TemplateParameter)[] templateList;
+		if (const tp = templateParameters)
+		if (const tpl = tp.templateParameterList)
+			templateList = tpl.items;
+		string[] tlList;
+		tlList = templateList.map!(a => templateParamName(a)).array;
+
+		// make a copy of all parameters and remove the seen ones later during the loop
+		const(TemplateParameter)[] tpList;
+		if (const tp = templateParameters)
+		if (const tpl = tp.templateParameterList)
+			tpList = tpl.items;
```

Comment:
`tpList` could be assigned at the same time as `templateList`. Also too bad it cant be copied (dup)

Reply:
with Rebindable you could use `.dup`:  `Rebindable!(const TemplateParameter) tpList, templateList`;

Reply:
`Rebindable` seems to be an alias for arrays and sadly doesn't work: 

```
src/dscanner/analysis/properly_documented_public_functions.d(208): Error: template object.dup cannot deduce function from argument types !()(const(TemplateParameter)[]), candidates are:
/usr/include/dlang/dmd/object.d(2086):        object.dup(T : V[K], K, V)(T aa)
/usr/include/dlang/dmd/object.d(2122):        object.dup(T : V[K], K, V)(T* aa)
/usr/include/dlang/dmd/object.d(4191):        object.dup(T)(T[] a) if (!is(const(T) : T))
/usr/include/dlang/dmd/object.d(4207):        object.dup(T)(const(T)[] a) if (is(const(T) : T))
```

Reply:
strange because i tried earlier. I even asked on IRC.

Reply:
[Look at this small runnable](https://run.dlang.io/is/xGE1FL)

Reply:
Interesting, but `remove` would still not as `remove` requires mutability of all elements (kind of stupid that you can't delete `const` elements, but well...)